feat(executions): Allow unqueuing to states other than RUNNING #5939

Malaydewangan09 opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?
Allows queued executions to be moved to RUNNING (default), CANCELLED, or FAILED states when unqueued.

closes #5939

Malaydewangan09 committed:
feat(executions): Allow unqueuing to states other than RUNNING #5939

# Conflicts:
#	webserver/src/main/java/io/kestra/webserver/controllers/api/ExecutionController.java

loicmathieu commented on webserver/src/main/java/io/kestra/webserver/controllers/api/ExecutionController.java:
```diff
@@ -1735,14 +1734,15 @@ public HttpResponse<?> setLabelsByQuery(
     public Execution unqueue(
-        @Parameter(description = "The execution id") @PathVariable String executionId
+        @Parameter(description = "The execution id") @PathVariable String executionId,
+        @Parameter(description = "The new state of the execution") @Nullable @QueryValue State.Type status
```
```suggestion
        @Parameter(description = "The new state of the execution") @Nullable @QueryValue State.Type state
```

Please use `state` anywhere and not `status`

loicmathieu reviewed (changes requested):
Please, also add a test that unqueue to a different state that the default.
And, well, it misses the frontend part?

Malaydewangan09 commented:
I’ll add the test. For the UI part, I might not be much help 🙃

Malaydewangan09 committed:
test(executions): add test for unqueuing to non-default states

anna-geller commented:
@loicmathieu the test is added, looks ready, maybe open a follow up issue for the UI change?

loicmathieu commented:
@anna-geller no, we didn't merge half-done PR, they have a hight risk of not being finished. I'll ask internally to find someone to amend the PR with the UI part.

MilosPaunovic committed:
feat(executions): added the ui part for being able to unqueue exeution to different state

MilosPaunovic committed:
feat(executions): passing state parameter to endpoint via query string parameters instead of body

MilosPaunovic committed:
chore(ui): pass default error message if there are no error logs sent from server

MilosPaunovic committed:
feat(executions): added the ui part for being able to unqueue exeution to different state from main listing

Malaydewangan09 committed:
fix: review changes

# Conflicts:
#	webserver/src/main/java/io/kestra/webserver/controllers/api/ExecutionController.java

Malaydewangan09 commented:
@loicmathieu
To confirm, now I need to change the `JdbcConcurrencyLimitService` ?

loicmathieu commented:
@Malaydewangan09 what you did should be enought.
If not, I may have a look.

Malaydewangan09 commented:
There's a compilation error in `JdbcConcurrencyLimitService `, since with my changes the signature of method has changed.

loicmathieu commented:
@Malaydewangan09 so yes, you need to impact it

Malaydewangan09 committed:
feat: Allow unqueuing to states other than RUNNING (JdbcConcurrencyLimitService)

loicmathieu reviewed (approved):
LGTM.

Please open a PR in EE for the io.kestra.ee.runner.kafka.services.KafkaConcurrencyLimitService to adapt it

Malaydewangan09 commented:
@loicmathieu I've made changes in `EE`, just to confirm we need to add the other two states i.e `CANCELLED, FAILED` in `ExecutionRunning.ConcurrencyState` enum so we can emit them with the correct state in the queue ?

loicmathieu commented:
@Malaydewangan09 I think so, please test it if possible

MilosPaunovic commented:
Hey @Malaydewangan09, don't forget about this one - keeping it opened for long will introduce more and more merge conflicts. It might be a good idea to wrap it up if it's close to completion?

MilosPaunovic commented:
@Malaydewangan09 You can ping me to resolve the conflicts on the UI together, when this get worked on.

Malaydewangan09 commented:
Sure, @MilosPaunovic 
I'll take this, once I am a bit done with plugins `P1` issues.

Malaydewangan09 committed:
feat(executions): Allow unqueuing to states other than RUNNING

# Conflicts:
#	ui/src/components/executions/Executions.vue

Malaydewangan09 committed:
feat(executions): Allow unqueuing to states other than RUNNING

Malaydewangan09 committed:
feat(executions): Allow unqueuing to states other than RUNNING

Malaydewangan09 committed:
feat(executions): Allow unqueuing to states other than RUNNING

MilosPaunovic reviewed (approved):
Seems A-OK from the side of the UI.

loicmathieu reviewed (approved):
LGTM but it would need changes on EE also, @Malaydewangan09 can you take care of them?

Malaydewangan09 commented:
@loicmathieu, I already added a PR [here](https://github.com/kestra-io/kestra-ee/pull/4348) and added you for review.
